baybe/searchspace/continuous.py

```diff
@@ -383,14 +408,17 @@ def sample_uniform(self, batch_size: int = 1) -> pd.DataFrame:
 
         if not self.parameters:
             return pd.DataFrame(index=pd.RangeIndex(0, batch_size))
-
-        if (
-            len(self.constraints_lin_eq) == 0
-            and len(self.constraints_lin_ineq) == 0
-            and len(self.constraints_cardinality) == 0
-        ):
+        # If the space is completely unconstrained, we can sample from bounds.
+        if not self.is_constrained:
             return self._sample_from_bounds(batch_size, self.comp_rep_bounds.values)
 
+        if self.has_interpoint_constraints:
+            return self._sample_from_polytope_with_interpoint_constraints(
```

Comment:
isnt 'sample_from_polytope' a special case of the new function? If so I think it could be better design if one was contained int he other or one calls the other

Reply:
You are right, I have combined the two functions into a single one and will resolve this comment here once the changes have been pushed- I will keep the other one open for further discussion of the topic.